### PR TITLE
adds expandable table row component

### PR DESCRIPTION
**Pull request by fpigeonjr**
## What changed

- adds composable expandable row component
- refactors Tables ExpandableRow component
- adds AllBLIRow component

Does not include: 
- edit/delete button functions
- e2e tests

## Issue

#1407 

## How to test

1. Goto Budgetlines tab
2. expand a row in the table
3. 💰profit

## Screenshots

image

### CODE REVIEW

File: frontend/src/pages/budgetLines/list/BudgetLineItemList.jsx [hunk 1]
```diff
@@ -68,10 +68,23 @@ export const BudgetLineItemList = () => {
     const budgetLinesWithCanAndAgreementName = budgetLineItems.map((budgetLine) => {
         const can = cans.find((can) => can.id === budgetLine.can_id);
         const agreement = agreements.find((agreement) => agreement.id === budgetLine.agreement_id);
+        const isLoggedInUserTheProjectOfficer = agreement.project_officer === loggedInUserId;
+        const isLoggedInUserTheAgreementCreator = agreement?.created_by === loggedInUserId;
+        const isLoggedInUserATeamMember = agreement?.team_members?.some(
+            (teamMember) => teamMember.id === loggedInUserId
+        );
+        const isLoggedInUserAllowedToEdit =
```
Comment (fpigeonjr) [inline frontend/src/pages/budgetLines/list/BudgetLineItemList.jsx:76]: this is a good candidate for a custom hook — re-usable business logic

File: frontend/src/components/UI/AllBLIRow/AllBLIRow.js [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+
+/**
+ * BLIRow component that represents a single row in the Budget Lines table.
+ * @param {Object} props - The props for the BLIRow component.
+ * @param {Object} props.bl - The budget line object.
+ * @param {boolean} [props.canUserEditBudgetLines] - Whether the user can edit budget lines.
+ * @param {boolean} [props.isReviewMode] - Whether the user is in review mode.
+ * @param {Function} [props.handleSetBudgetLineForEditing] - The function to set the budget line for editing.
+ * @param {Function} [props.handleDeleteBudgetLine] - The function to delete the budget line.
+ * @param {Function} [props.handleDuplicateBudgetLine] - The function to duplicate the budget line.
```
Comment (johndeange) [inline frontend/src/components/UI/AllBLIRow/AllBLIRow.js:27]: not a prop

File: frontend/src/components/UI/AllBLIRow/AllBLIRow.js [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+    totalBudgetLineFeeAmount,
+    totalBudgetLineAmountPlusFees,
+} from "../../../helpers/utils";
+
+/**
+ * BLIRow component that represents a single row in the Budget Lines table.
+ * @param {Object} props - The props for the BLIRow component.
+ * @param {Object} props.bl - The budget line object.
+ * @param {boolean} [props.canUserEditBudgetLines] - Whether the user can edit budget lines.
+ * @param {boolean} [props.isReviewMode] - Whether the user is in review mode.
```
Comment (johndeange) [inline frontend/src/components/UI/AllBLIRow/AllBLIRow.js:24]: not a prop

File: frontend/src/components/UI/AllBLIRow/AllBLIRow.js [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+    handleSetBudgetLineForEditing = () => {},
+    handleDeleteBudgetLine = () => {},
+    readOnly = false,
+}) => {
+    const [isExpanded, setIsExpanded] = React.useState(false);
+    const [isRowActive, setIsRowActive] = React.useState(false);
+    let loggedInUser = useSelector((state) => loggedInName(state?.auth?.activeUser));
+    const loggedInUserId = useSelector((state) => state?.auth?.activeUser?.id);
+
+    const isBudgetLineDraft = budgetLine?.status === "DRAFT";
```
Comment (johndeange) [inline frontend/src/components/UI/AllBLIRow/AllBLIRow.js:43]: would be nice to create a reusable function for this (I think it's implemented on a few components now)

**Review by johndeange — APPROVED** [review]

**Review by stevtek — APPROVED** [review]

**Pull request merged**